### PR TITLE
[PBI-1953]

### PR DESCRIPTION
**Pull request by aeng7**
Conviva:
-added support for applicationName and streamUrl
-added unit tests for applicationName and streamUrl

### CODE REVIEW

File: js/plugins/conviva.js [hunk 1]
```diff
@@ -246,7 +247,11 @@ var ConvivaAnalyticsPlugin = function(framework)
       // A human-readable identifier for your application.
       // Very helpful to filter traffic and compare performance for different builds of
       // the video application.
-      // contentMetadata.applicationName = "JS SDK Sample Application 1.2.3";
+      var appName = convivaMetadata["applicationName"];
+      if (_.isString(appName))
+      {
+        contentMetadata.applicationName = appName;
```
Comment (pilievOoyala) [inline js/plugins/conviva.js]: Is there a default for contentMetadata.applicationName?  Should it just equal the empty string?

Comment (aeng7) [inline js/plugins/conviva.js]: It's not a required field by the Conviva SDK, but it is a required field for the SkyNews metadata we will be testing against.

File: js/plugins/conviva.js [hunk 2]
```diff
@@ -500,6 +512,7 @@ var ConvivaAnalyticsPlugin = function(framework)
    */
   var resetContentState = function()
   {
+    streamUrl = null;
```
Comment (pilievOoyala) [inline js/plugins/conviva.js]: does streamType need to be cleared? What about currentConvivaSessionKey?

Comment (aeng7) [inline js/plugins/conviva.js]: currentConvivaSessionKey is set to the NO_SESSION_KEY value in the clearLastSession function. streamType we will need to double check how often it updates.

**Comment by pilievOoyala** [comment]
LGTM

**Pull request merged**